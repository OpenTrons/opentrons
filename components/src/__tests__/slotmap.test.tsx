// slot map  component tests
import { SlotMap } from '..'
import React from 'react'
import Renderer from 'react-test-renderer'

describe('SlotMap', () => {
  it('renders correctly without collision warnings or errors', () => {
    const tree = Renderer.create(<SlotMap occupiedSlots={['1']} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with collision warning', () => {
    const tree = Renderer.create(
      <SlotMap occupiedSlots={['1']} collisionSlots={['4']} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with error', () => {
    const tree = Renderer.create(
      <SlotMap occupiedSlots={['1']} isError />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with error and collision warning', () => {
    const tree = Renderer.create(
      <SlotMap occupiedSlots={['1']} collisionSlots={['4']} isError />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
