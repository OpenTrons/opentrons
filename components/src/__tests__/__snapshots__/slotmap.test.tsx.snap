// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotMap renders correctly with collision warning 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<svg
  viewBox="-1,-1,101, 94"
>
  <g
    transform="translate(0 0)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 0)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 23)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 23)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 23)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 46)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
    <svg
      aria-hidden="true"
      className="sc-AxmLO c0 collision_icon"
      fill="currentColor"
      height={20}
      version="1.1"
      viewBox="0 0 24 24"
      width={20}
      x={6.5}
      xmlns="http://www.w3.org/2000/svg"
      y={1.5}
    >
      <path
        d="M13 9h-2V7h2m0 10h-2v-6h2m-1-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10 10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
        fillRule="evenodd"
      />
    </svg>
  </g>
  <g
    transform="translate(33 46)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 46)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 69)"
  >
    <rect
      className="slot_rect slot_occupied"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 69)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 69)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
</svg>
`;

exports[`SlotMap renders correctly with error 1`] = `
<svg
  viewBox="-1,-1,101, 94"
>
  <g
    transform="translate(0 0)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 0)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 23)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 23)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 23)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 46)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 46)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 46)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 69)"
  >
    <rect
      className="slot_rect slot_occupied slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 69)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 69)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
</svg>
`;

exports[`SlotMap renders correctly with error and collision warning 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<svg
  viewBox="-1,-1,101, 94"
>
  <g
    transform="translate(0 0)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 0)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 23)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 23)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 23)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 46)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
    <svg
      aria-hidden="true"
      className="sc-AxmLO c0 collision_icon"
      fill="currentColor"
      height={20}
      version="1.1"
      viewBox="0 0 24 24"
      width={20}
      x={6.5}
      xmlns="http://www.w3.org/2000/svg"
      y={1.5}
    >
      <path
        d="M13 9h-2V7h2m0 10h-2v-6h2m-1-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10 10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
        fillRule="evenodd"
      />
    </svg>
  </g>
  <g
    transform="translate(33 46)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 46)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 69)"
  >
    <rect
      className="slot_rect slot_occupied slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 69)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 69)"
  >
    <rect
      className="slot_rect slot_error"
      height={23}
      width={33}
    />
  </g>
</svg>
`;

exports[`SlotMap renders correctly without collision warnings or errors 1`] = `
<svg
  viewBox="-1,-1,101, 94"
>
  <g
    transform="translate(0 0)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 0)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 23)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 23)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 23)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 46)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 46)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 46)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(0 69)"
  >
    <rect
      className="slot_rect slot_occupied"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(33 69)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
  <g
    transform="translate(66 69)"
  >
    <rect
      className="slot_rect"
      height={23}
      width={33}
    />
  </g>
</svg>
`;
